src/modules/integrations/hostctl.nix

```diff
@@ -0,0 +1,46 @@
+{ pkgs, lib, config, ... }:
+
+let
+  hostContent = lib.concatStringsSep "\n" config.hostctl.hosts;
+  hostHash = builtins.hashString "sha256" hostContent;
+  file = pkgs.writeText "hosts" ''
+    # ${hostHash}
+    ${hostContent}
+  '';
+in
+{
+  options.hostctl = {
+    enable = lib.mkEnableOption "Adjust /etc/hosts using hostctl.";
+
+    profile = lib.mkOption {
+      type = lib.types.str;
+      default = "devenv";
+      description = "Profile name to use.";
+    };
+
+    hosts = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = "List of hosts entries.";
+    };
+  };
+
+  config = lib.mkIf config.hostctl.enable {
```

Comment:
Using lib.mkMerge (see nixpkgs for examples) you can also handle when it's not enabled and remove the profile if it exists.